pkg/apis/machine/types.go

```diff
@@ -492,10 +496,27 @@ const (
 
 	// RollingUpdateMachineDeploymentStrategyType means that old MCs will be replaced by new one using rolling update i.e gradually scale down the old MCs and scale up the new one.
 	RollingUpdateMachineDeploymentStrategyType MachineDeploymentStrategyType = "RollingUpdate"
+
+	// InPlaceUpdateMachineDeploymentStrategyType signifies that machines will be updated in place,
+	// gradually transitioning from the old MachineSet to the new MachineSet without requiring machine recreation.
+	InPlaceUpdateMachineDeploymentStrategyType MachineDeploymentStrategyType = "InPlaceUpdate"
 )
 
 // RollingUpdateMachineDeployment specifies the spec to control the desired behavior of rolling update.
 type RollingUpdateMachineDeployment struct {
+	UpdateConfiguration
+}
+
+// InPlaceUpdateMachineDeployment specifies the spec to control the desired behavior of inplace rolling update.
+type InPlaceUpdateMachineDeployment struct {
+	UpdateConfiguration
+
+	// OrchestrationType specifies the orchestration type for the inplace update.
+	OrchestrationType OrchestrationType
```

Comment:
How about moving this inside `UpdateConfiguration` itself ?  In the future we could have ManualRolling as well

Reply:
I would prefer the current API, as there is no plan also in the long run of ManualRolling.

Reply:
FIne with me.